performancecopilot/pcp
# qa, vm setup, small code changes to remove warnings Changes committed to git@github.com:kmcdonell/pcp.git 20190829

Ken McDonell (9):
      qa/914 & qa/src/profilecrash.c: generalize metric use
      qa/common.config: QA Farm subnet changed
      qa/914: speed up
      src/libpcp/src/getdate.y: for pure parser use a more modern declaration
      qa/src/getdomainname.c: fix compilation warning
      qa/README.vm: OpenIndiana updates for pkg use
      qa/admin/check-vm: improvement for unknown packages
      qa/admin/other-packages/manifest: OpenIndiana changes
      qa/admin/other-packages/unavailable: updates for assorted platforms

 qa/914                              |   35 ++++++++++++++----
 qa/914.out                          |    6 +--
 qa/README.vm                        |    6 ++-
 qa/admin/check-vm                   |    6 ++-
 qa/admin/other-packages/manifest    |    4 +-
 qa/admin/other-packages/unavailable |    6 ++-
 qa/common.config                    |    8 ++--
 qa/src/getdomainname.c              |    2 +
 qa/src/profilecrash.c               |   70 +++++++++++++++++++++++++++++-------
 src/libpcp/src/getdate.y            |    8 ++++
 10 files changed, 119 insertions(+), 32 deletions(-)

Details ...

commit fc6151b382f15bdf4839fb0b4f2f33b0adbce983
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Aug 29 09:00:19 2019 +1000

    qa/admin/other-packages/unavailable: updates for assorted platforms

commit 908eec06ec6e346126fe9a8c2a95101628b9449a
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Aug 29 09:00:05 2019 +1000

    qa/admin/other-packages/manifest: OpenIndiana changes

commit c399cd853622db365e4f3c64a3237d87c3d9757e
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Aug 29 08:58:00 2019 +1000

    qa/admin/check-vm: improvement for unknown packages
    
    For platforms where the package (or N/A status) is unknown we
    tend to use
            ... [? ...]
    in the manifest.
    
    This change allows check-vm -p to report ?:<line number> instead
    of ? so these can more easily be found and fixed in the manifest.

commit e0d32d4b2f6923187ca03c0aad8db21755c1b352
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Aug 29 08:57:36 2019 +1000

    qa/README.vm: OpenIndiana updates for pkg use

commit 82a63ca2443bf1a03fba1fa9b25403e2fbe0d3f2
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Aug 29 08:55:50 2019 +1000

    qa/src/getdomainname.c: fix compilation warning
    
    (when getdomainname() not available, like on OpenIndiana)

commit 2ecf859d7b43c4b183cfae1773543842cfb7593d
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Aug 29 08:54:48 2019 +1000

    src/libpcp/src/getdate.y: for pure parser use a more modern declaration
    
    Compilation warning from bison spotted on OpenIndiana.

commit 9b8b16864f3b1cd45058da5591a3c922eb5dcc30
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Aug 29 08:52:47 2019 +1000

    qa/914: speed up
    
    The .xz archive is big ... on small memory VMs we do a lot of slow
    decompressing, so decompress once and use the temporary decompressed
    archive with src/profilecrash.

commit d79ea2499cc1b44b14ae96a45e5bacca17c90d20
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Aug 29 08:33:01 2019 +1000

    qa/common.config: QA Farm subnet changed
    
    Don't even start me on the pox that is the NBN in Australia.

commit f4ddd53f1bbdf5ecacd2fb5f7d7e08caa848a2cf
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Thu Aug 29 08:31:10 2019 +1000

    qa/914 & qa/src/profilecrash.c: generalize metric use
    
    Use sampledso.hordes.one instead of (the rarely available) NFS
    server stats.
